garmin_txt.cc

```diff
@@ -1304,7 +1311,7 @@ parse_track_waypoint()
       }
       break;
     case 9:
-      WAYPT_SET(wpt, course, atoi(str));
+      WAYPT_SET(wpt, course, atoi(CSTR(str)));
```

Comment:
Is this str.toDouble() ?

Reply:
The converstion to QString::toTYPE requires knowledge that the value being converted does not have a trailing non-convertable part.  atoTYPE will ignore trailing non-convertable data while QString::toTYPE will not.